### PR TITLE
Nazar, Maryna, first CRUDs

### PR DESCRIPTION
**Pull request by Nazar-kob**

### CODE REVIEW

File: app/views/illness.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+@login_required
+def detail(illness_id: int):
+    form = f.IllnessForm()
+
+    if form.name.data and db.session.scalar(
+        sa.Select(m.Illness.name).where(m.Illness.name == form.name.data, m.Illness.id != illness_id)
+    ):
+        log(log.INFO, "Illness name already exist! [%s]", form.name.data)
+        flash("Illness name already exist!", "danger")
+        return redirect(url_for("illness.get_all"))
```
Comment (Taraslysak) [inline app/views/illness.py]: Shouldn't we do these checks only in case of `validate_on_submit` returns `True`?

File: app/views/illness.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    return render_template("illness/modal_form.html", form=form, illness_id=illness_id)
+
+
+@bp.route("/create", methods=["GET", "POST"])
+@login_required
+def create():
+    form = f.IllnessForm()
+
+    if form.name.data and db.session.scalar(sa.Select(m.PlantFamily.name).where(m.PlantFamily.name == form.name.data)):
```
Comment (Taraslysak) [inline app/views/illness.py]: Again, shouldn't we validate form first?

File: app/views/pest.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        query = m.Pest.select().where(m.Pest.id == int(form.pest_id.data))
+        pest: m.Pest | None = db.session.scalar(query)
+        if not pest or pest.is_deleted:
+            log(log.ERROR, "Not found pest by id : [%s]", form.pest_id.data)
+            flash("Cannot save pest data", "danger")
+            return redirect(url_for("pest.get_all"))
+        pest.name = form.name.data
+        pest.symptoms = form.symptoms.data
+        pest.treatment = form.treatment.data
+        # pest.photos = form.photos.data
```
Comment (Taraslysak) [inline app/views/pest.py:67]: redundant comment?

File: app/views/pest.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        log(log.INFO, "Pest name already exist! [%s]", form.name.data)
+        flash("Pest name already exist!", "danger")
+        return redirect(url_for("pest.get_all"))
+
+    if form.validate_on_submit():
+        pest = m.Pest(
+            name=form.name.data,
+            symptoms=form.symptoms.data,
+            treatment=form.treatment.data,
+            # photos=form.photos.data,
```
Comment (Taraslysak) [inline app/views/pest.py:92]: redundant comment?

File: app/views/plant_family.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+
+bp = Blueprint("plant_family", __name__, url_prefix="/plant-family")
+
+
+@bp.route("/", methods=["GET"])
+@login_required
+def get_all():
+    q = request.args.get("q", type=str, default=None)
+    where = sa.and_(m.PlantFamily.is_deleted.is_(False))
```
Comment (Taraslysak) [inline app/views/plant_family.py:25]: Shouldn't `and_` accept 2 arguments?

File: app/views/plant_family.py [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+        search_query=q,
+    )
+
+
+@bp.route("/create", methods=["GET", "POST"])
+@login_required
+def create():
+    form = f.PlantFamilyForm()
+    form.pests.choices = db.session.scalars(sa.Select(m.Pest.name).where(m.Pest.is_deleted.is_(False))).all()
+    form.illnesses.choices = db.session.scalars(sa.Select(m.Illness.name).where(m.Illness.is_deleted.is_(False))).all()
```
Comment (Taraslysak) [inline app/views/plant_family.py:49]: I assume we do these queries only when we receive "GET" method? If we receive "POST" request with form these form fields should be pre-occupied, isn't it?

File: app/views/plant_family.py [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+    if form.name.data and db.session.scalar(sa.Select(m.PlantFamily.name).where(m.PlantFamily.name == form.name.data)):
+        log(log.INFO, "PlantFamily name already exist! [%s]", form.name.data)
+        flash("Plan Family name already exist!", "danger")
+        return redirect(url_for("plant_family.get_all"))
+
+    if request.method == "POST" and form.validate_on_submit():
+        pests = db.session.scalars(sa.Select(m.Pest).where(m.Pest.name.in_(form.pests.data)))
+        illness = db.session.scalars(sa.Select(m.Illness).where(m.Illness.name.in_(form.pests.data)))
```
Comment (Taraslysak) [inline app/views/plant_family.py:58]: May be we need to check for `is_deleted == False` here also?

File: app/views/plant_variety.py [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+#         if form.errors:
+#             log(log.INFO, "Form error Plant family! [%s]", form.errors)
+#             flash(f"{form.errors}", "danger")
+#         return redirect(url_for("plant_family.get_all"))
+
+#     form.name.data = plant_family.name
+#     form.features.data = plant_family.features
+#     form.pests.data = [pest.name for pest in plant_family.pests]
+#     form.illnesses.data = [illness.name for illness in plant_family.illnesses]
+#     return render_template("plant_family/modal_form.html", form=form, plant_family_id=plant_family.id)
```
Comment (Taraslysak) [inline app/views/plant_variety.py:115]: redundant comments?

**Review by Taraslysak — COMMENTED** [review]
Please consider minor refactoring

File: app/views/plant_family.py (hunk 6 above)
Comment (Nazar-kob) [reply app/views/plant_family.py:49]: done

**Pull request merged**